xarray/core/pycompat.py

```diff
@@ -45,3 +45,13 @@ def is_duck_dask_array(x):
 except ImportError:  # pragma: no cover
     cupy_version = LooseVersion("0.0.0")
     cupy_array_type = ()
+
+try:
+    # solely for isinstance checks
+    import pint
+
+    pint_version = LooseVersion(pint.__version__)
+    pint_array_type = (pint.Quantity,)
```

Comment:
Do you think this explains the failures in #5568 too?

Reply:
Looking at this properly I see what you mean now. So is this PR just stuck then? Why can we not just check the name rather than the explicit type within xarray (for now at least)?

Reply:
you could import `pint` in the functions that use `pint_array_type` for now, and I'll try to get `pint` to not import `xarray` (because ideally it should not care about what wraps it). The best long term fix would be to figure out dask/dask#6635, though.

Reply:
@keewis It would be good to hear your thoughts on that over on the Pint issue tracker. Right now, to maintain the type casting hierarchy, libraries with duck arrays have to either define a list of types they can wrap (and prohibit all others) or a list which are meant to wrap it (and allow wrapping any other). Pint went with the latter since it was the safest approach as a type high on the graph (https://pint.readthedocs.io/en/stable/numpy.html#Technical-Commentary), but I suppose that if that was problematic, it could be flipped around to a list of allowed types to wrap (with a breaking change), as Dask ended up doing (https://github.com/dask/dask/pull/6393).

But, definitely, this ad hoc approach to the type casting hierarchy will continue to be problematic until https://github.com/dask/dask/issues/6635 is resolved definitely.

Reply:
It would be really nice to fix this properly (and I will bring it up in the xarray dev call now), but for this PR then your suggestion seems to work nicely @keewis , thanks.